### PR TITLE
[TT-3154] Strip listen path when it is regex too

### LINKED ISSUE #2249: Stripping listen paths does not work if listen path contains regexp
**Do you want to request a *feature* or report a *bug*?**
bug
**What is the current behavior?**
If I define listen path like `/{_:.*}/listen`, it works as expected, except stripping listen paths. 
At the moment it can strip only "static" values, and not handle regexps. 
**What is the expected behavior?**
It should detect if regexp is defined, and strip the path using regexp.
**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem**
See above
**Which versions of Tyk affected by this issue? Did this work in previous versions of Tyk?**
All till 2.8

### PR DESCRIPTION
**Pull request by furkansenharputlu**
Fixes #2249

### CODE REVIEW

File: handler_error.go [hunk 1]
```diff
@@ -94,7 +94,11 @@ func (e *ErrorHandler) HandleError(w http.ResponseWriter, r *http.Request, errMs
 		}
 
 		if e.Spec.Proxy.StripListenPath {
-			r.URL.Path = strings.Replace(r.URL.Path, e.Spec.Proxy.ListenPath, "", 1)
```
Comment (buger) [inline handler_error.go]: Are you sure that RawPath should be changed here?
If so does it mean that you solved some bug?

File: handler_success.go [hunk 2]
```diff
@@ -332,6 +338,21 @@ func (s *SuccessHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) *http
 	return nil
 }
 
+func stripListenPath(r *http.Request, listenPath string) error {
+	tmp := new(mux.Route).PathPrefix(listenPath)
```
Comment (buger) [inline handler_success.go]: Not quite sure if understand how it works, can you add a comment?
Also, I feel like Regexp can be compiled only once, and cached somewhere in API def.

File: gateway/handler_success.go [hunk 3]
```diff
@@ -332,6 +338,21 @@ func (s *SuccessHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) *http
...
 }
 
+func stripListenPath(r *http.Request, listenPath string) error {
+	tmp := new(mux.Route).PathPrefix(listenPath)
+	s, err := tmp.GetPathRegexp()
+	if err != nil {
+		return err
+	}
+
+	reg := regexp.MustCompile(s)
```
Comment (matiasinsaurralde) [inline gateway/handler_success.go]: Not sure about this particular scenario but our `regexp` package might be useful here? https://github.com/TykTechnologies/tyk/blob/master/regexp/regexp.go

cc @dencoded

Comment (buger) [reply gateway/handler_success.go]: It is already used, see above

**Comment by buger** [comment]
@furkansenharputlu this can be very slow. We at least need to store cached regexp variable in API def package.

**Comment by furkansenharputlu** [comment]
Not stale

**Comment by lghiur** [comment]
This issue has been addressed under this PR https://github.com/TykTechnologies/tyk/pull/4104

**Pull request closed without merging**